### ISSUE TITLE
"transmission-remote -l" fails with segmentation fault

### ISSUE THREAD

**Issue by fredo-47** [transmission/transmission#5958]
### What is the issue?

I run transmission daemon on my NAS (Linux). 
Mostly I connect to it with transgui, but sometimes I use transmission-remote.

Now with the latest versions, when I ask for the list of all torrents via `transmission-remote -l`, I get a segmentation fault:
`$ transmission-remote -l`
`    ID   Done       Have  ETA            Up    Down  Ratio  Status       Name`
`Segmentation fault (core dumped)`

Btw, when there are no torrents, then `transmission-remote -l` doesn't produce a segmentation fault.
And doing other queries, e.g. `transmission -t 1 -i`, work fine and deliver the normal expected return.

The same happens when I do this on my desktop machine (Linux Mint) with has the previous version cb2462aacd.
Same with 9df4adf9b63dae6ef5540903a3dbf0a70bafbfbf

I'm not sure if the cause is in t-remote or t-daemon. 
Of course, it could be that something on my system has changed?

Any ideas on how to look deeper into this?

### Which application of Transmission?

transmission-daemon

### Which version of Transmission?

fc4b7ed9f4

**Comment by ckerr** [comment]
Looks like this a regression introduced by my recent `tr_variant` refactors. Thanks for reporting it @fredo-47 :bow: 

This is only in `main` and doesn't affect `4.0.x`.

**Comment by ckerr** [comment]
```
/home/charles/src/transmission/libtransmission/variant.cc:172:9: runtime error: store to null pointer of type 'int64_t' (aka 'long')
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior /home/charles/src/transmission/libtransmission/variant.cc:172:9 in 
AddressSanitizer:DEADLYSIGNAL
=================================================================
==299510==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000000 (pc 0x55ed7aff9d5b bp 0x7ffdb90dcfa0 sp 0x7ffdb90dcf20 T0)
==299510==The signal is caused by a WRITE memory access.
==299510==Hint: address points to the zero page.
    #0 0x55ed7aff9d5b in tr_variantGetInt(tr_variant const*, long*) /home/charles/src/transmission/libtransmission/variant.cc:172:16
    #1 0x55ed7affb5b9 in tr_variantDictFindInt(tr_variant*, unsigned long, long*) /home/charles/src/transmission/libtransmission/variant.cc:285:12
    #2 0x55ed7a9d6d06 in printTorrentList(tr_variant*) /home/charles/src/transmission/utils/remote.cc:1485:17
    #3 0x55ed7a9944df in processResponse(char const*, std::basic_string_view<char, std::char_traits<char>>, Config&) /home/charles/src/transmission/utils/remote.cc:2205:21
    #4 0x55ed7a98a181 in flush(char const*, tr_variant*, Config&) /home/charles/src/transmission/utils/remote.cc:2368:23
    #5 0x55ed7a98a259 in flush(char const*, tr_variant*, Config&) /home/charles/src/transmission/utils/remote.cc:2380:23
    #6 0x55ed7a97e976 in processArgs(char const*, int, char const* const*, Config&) /home/charles/src/transmission/utils/remote.cc:2648:23
    #7 0x55ed7a975bfb in main /home/charles/src/transmission/utils/remote.cc:3348:12
    #8 0x7f205d229d8f in __libc_start_call_main csu/../sysdeps/nptl/libc_start_call_main.h:58:16
    #9 0x7f205d229e3f in __libc_start_main csu/../csu/libc-start.c:392:3
    #10 0x55ed7a89a754 in _start (/home/charles/src/transmission/build/utils/transmission-remote+0x1001754) (BuildId: 7fd1c4c539fbab23b9bee24bd07984570f2807d7)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV /home/charles/src/transmission/libtransmission/variant.cc:172:16 in tr_variantGetInt(tr_variant const*, long*)
==299510==ABORTING
```

**Comment by ckerr** [comment]
Fixed in #5960.
